paasta_tools/setup_marathon_job.py

```diff
@@ -242,6 +243,11 @@ def log_bounce_action(line, level='debug'):
 
     kill_given_tasks(client=client, task_ids=[task.id for task in tasks_to_kill], scale=True)
 
+    for task in bounce_lib.flatten_tasks(old_app_at_risk_tasks):
+        if task in tasks_to_kill:
+            hostname = task.host
+            reserve_all_resources([hostname])
```

Comment:
The `undrain()` function will call `unreserve_all_resources()`. We will need to make sure things like the autoscaler and other tools remember to call `undrain()`.